core/src/main/java/io/grpc/ClientCall.java

```diff
@@ -38,7 +38,8 @@
  * Applications are expected to utilize normal payload messages for such signals, as a response
  * naturally acknowledges its request.
  *
- * <p>Methods are guaranteed to be non-blocking. Implementations are not required to be thread-safe.
+ * <p>Methods are guaranteed to be non-blocking. Implementations are not required to be thread-safe
+ * except for {@link #request}, which may be called from any thread.
```

Comment:
IMO Javadoc is facing the API users not the implementors, so `Implementations are not required ...` should not be mentioned here in the javadoc at all. It can be still be added in form of an ordinary comment `//...` or `/* ... */` out side the javadoc, or use the javadoc tag `@implNote` (require java 8).

Reply:
Or rephrase it to be user-facing like `Not thread-safe except for  {@link #request}, which may be called from any thread.`

Reply:
Reprhased. I did not clean up other examples of this kind of phrasing in this file, though.